processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -107,12 +108,14 @@ public SegmentMetadataQueryQueryToolChest(SegmentMetadataQueryConfig config, Gen
           Map<String, Object> context
       )
       {
-        Query<SegmentAnalysis> query = queryPlus.getQuery();
+        SegmentMetadataQuery castedQuery = (SegmentMetadataQuery) queryPlus.getQuery();
+        SegmentMetadataQuery updatedQuery =(SegmentMetadataQuery) (castedQuery.withAnalysisTypes(getFinalAnalysisTypes(castedQuery)));
```

Comment:
Suggested to join those methods into a single method like `SegmentMetadataQuery.withFinalizedAnalysisTypes(toolChest)`, for less boilerplate, because currently they are always used together. Also this method may avoid creating a new query object and return the same instance, if analysisTypes are already set.